install/kubernetes/cilium/templates/spire/server/statefulset.yaml

```diff
@@ -72,6 +76,10 @@ spec:
             port: 8080
           initialDelaySeconds: 5
           periodSeconds: 5
+        {{- with .Values.authentication.mutual.spire.install.server.securityContext }}
+        securityContext:
+          {{- toYaml  . | nindent 10 }}
```

Comment:
Nit: there's a double space between `toYaml` and `.` on this line.

Reply:
Thanks, Done.  I am not sure why `:spelling:ignore:` is removed from `helm-values.rst`

Reply:
This is likely because your `main` branch is too old, and the Documentation Makefile does not contain the code to add these tags yet? Can you try to rebase your PR on the top of the current main (`git checkout main; git rebase main feat/enable-podsecuritycontext-on-spire-server`) and re-run the command, please?

Reply:
I have tried the following and yet not getting `:spelling:ignore:` prefix.
 
 1. Updated/synced my fork at https://github.com/ishuar/cilium/tree/main
 2.  Checkout main locally and ran `git pull`  , (latest commit is in the rebase output)
 3. ran `make -C Documentation update-helm-values` from the 


<details>
<summary>Click to check the local commands workflow</summary>


```bash
╰─$ git checkout main ; git pull ; git rebase main feat/enable-podsecuritycontext-on-spire-server
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
Already up to date.
Successfully rebased and updated refs/heads/feat/enable-podsecuritycontext-on-spire-server.
╰─$ make -C Documentation update-helm-values
docker container run --rm --workdir /src/Documentation --volume <PATH_TO_CILLIUM_FORKED_REPO>cilium/Documentation/..:/src --env READTHEDOCS_VERSION= --env SKIP_LINT= --user "501:20" "quay.io/cilium/helm-toolbox:"v1.0.1"@sha256:"a1483fc036b15c290326ba7ed36b8f0b714e185913b72a6074c04225d74e034c"" helm-docs -d -c /src/install/kubernetes -t /src/Documentation/helm-values.tmp.tmpl > helm-values.tmp
time="2023-08-14T13:53:14Z" level=info msg="Found Chart directories [cilium]"
time="2023-08-14T13:53:14Z" level=info msg="Generating README Documentation for chart /src/install/kubernetes/cilium"
awk -F'|' '{print "|"$2"|"$5"|"$3"|"$4"|"}' helm-values.tmp > helm-values.awk
docker container run --rm --workdir /src/Documentation --volume /Users/is/Documents/Personal-GitHub/cilium/Documentation/..:/src --env READTHEDOCS_VERSION= --env SKIP_LINT= --user "501:20" "quay.io/cilium/helm-toolbox:"v1.0.1"@sha256:"a1483fc036b15c290326ba7ed36b8f0b714e185913b72a6074c04225d74e034c"" python3 /usr/bin/m2r2 --overwrite helm-values.awk
sed 's/^\(   \* - \)\([[:print:]]\+\)$/\1:spelling:ignore:`\2`/' helm-values.rst > helm-values.sed
printf '..\n  %s\n\n%s\n' "AUTO-GENERATED. Please DO NOT edit manually." "$(cat helm-values.sed)" > helm-values.rst
rm -f -- helm-values.tmp helm-values.awk helm-values.sed
╰─$ git log --oneline --max-count=10                                                             
ed30c230fc (HEAD -> feat/enable-podsecuritycontext-on-spire-server) added the securityContext in Spire server pod and container.
f439a14d8e (origin/main, origin/HEAD, main) bpf: policy: fine-tune logic in __policy_can_access()
b113273df2 bpf: policy: fine-tune ICMP logic
ef7f6f4112 bpf: policy: add policy_can_ingress*() wrappers
1410a661c1 Add SPIRE connection to status
916a2ce2c2 health: only launch /hello after host datapath is ready
6c0d4787f0 Bump Golang to v1.20 on Cilium v1.12 and v1.13
92481c499b egressgw: back out test for policy conflict in ENI mode
3c366bd3b9 docs, cilium: Remove `cilium endpoint regenerate` command
c49bf315e6 hive: prevent goleak error due to race condition

```

</details>

Reply:
```
sed 's/^\(   \* - \)\([[:print:]]\+\)$/\1:spelling:ignore:`\2`/' helm-values.rst > helm-values.sed
printf '..\n  %s\n\n%s\n' "AUTO-GENERATED. Please DO NOT edit manually." "$(cat helm-values.sed)" > helm-values.rst
```

These are the commands that should take care of adding the tags indeed. Did you forget to  `git add` and `git commit` the changes after that? Or is the file still missing the role tags after you ran this command?

Reply:
@qmonnet I also have the same understanding of the commands. but seems like sed on macOS does not work as expected in the make command. 

I have the correct makefile locally as you can see in my execution it is running `sed 's/^\(   \* - \)\([[:print:]]\+\)$/\1:spelling:ignore:`\2`/' helm-values.rst > helm-values.sed` but not inserting the string `:spelling:ignore:` to the regex pattern.

Reply:
Oooh, macOS, I get it now. I tried to fix this command once for macOS already, but I don't have any Apple hardware around to test it, so I take it my fix was not good enough.

Would you mind helping me figure out the right command? I wonder if this is the `[[:print:]]` which is not supported on your version. What do you get with:

```
$ echo foo | sed 's/[[:print:]]/b/'    
boo
```

Alternatively, if you have GNU sed installed (`gsed`), the Makefile should pick it up over your regular `sed` and it should work. But it would be great to fix this command once and for all.

Reply:
I installed `gnu-sed` for verifying and yes it's the default `sed` in MacOS causing the issue.

```bash
╰─$ gsed  's/^\(   \* - \)\([[:print:]]\+\)$/\1:spelling:ignore:`\2`/' helm-values.rst > helm-values.sed | grep -i "spelling" | head -n 5
   * - :spelling:ignore:`Key`
   * - :spelling:ignore:`MTU`
   * - :spelling:ignore:`affinity`
   * - :spelling:ignore:`agent`
   * - :spelling:ignore:`agentNotReadyTaintKey`
  ```

Reply:
From https://riptutorial.com/sed/topic/9436/bsd-macos-sed-vs--gnu-sed-vs--the-posix-sed-specification, it's maybe the `\+` that causes the issue

Reply:
```bash
$ echo foo | sed 's/[[:print:]]/b/'  
boo
```

Reply:
`sed 's/^\(   \* - \)\([[:print:]]\{1,\}\)$/\1:spelling:ignore:`\2`/'` worked 🎆

Reply:
@qmonnet the above command is working fine on macos with default `sed` may be you could verify on other platforms if required.

Reply:
Great! Thanks for testing

> do you think I should create another PR for the helm docs fix for macOS or can it be part of this PR?

Ideally another PR, because we'll need to mark it for backport to all stable branches (unlike this PR).

Reply:
@qmonnet Thanks for your time and support , here is the new PR 

- https://github.com/cilium/cilium/pull/27495

Reply:
Thanks a lot! I'll take a look tomorrow.

Reply:
Now merged :+1: